pkg/apis/core/validation/validation.go

```diff
@@ -4053,6 +4053,35 @@ func ValidateService(service *core.Service) field.ErrorList {
 		ports[key] = true
 	}
 
+	// Validate TopologyKeys
+	if len(service.Spec.TopologyKeys) > 0 {
+		topoPath := specPath.Child("topologyKeys")
+		// topologyKeys is mutually exclusive with 'externalTrafficPolicy=Local'
+		if service.Spec.ExternalTrafficPolicy == core.ServiceExternalTrafficPolicyTypeLocal {
+			allErrs = append(allErrs, field.Forbidden(topoPath, "may not be specified when `externalTrafficPolicy=Local', and 'kubernetes.io/hostname' can be used as a topology key to replicate 'externalTrafficPolicy=Local'"))
```

Comment:
I'm not sure that the statement `'kubernetes.io/hostname' can be used as a topology key to replicate 'externalTrafficPolicy=Local'` is true yet. It should be the end goal though IMO. We probably shouldn't tell users until it's true?

Reply:
I think you're right, let's just delete it temporarily.